transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollServerChannel.java

```diff
@@ -75,11 +76,19 @@ public void connect(SocketAddress socketAddress, SocketAddress socketAddress2, C
         @Override
         void epollInReady() {
             assert eventLoop().inEventLoop();
+            boolean edgeTriggered = isFlagSet(Native.EPOLLET);
+
+            final ChannelConfig config = config();
+            if (!readPending && !config.isAutoRead() && !edgeTriggered) {
```

Comment:
It's actually used below two more times (@normanmaurer forgot to change [this line](https://github.com/netty/netty/pull/3442/files#diff-e069274ce2f565d9de425d5c35019dd2R111)). Perhaps we could do this:

```
final ChannelConfig config;
if (!readPending && !edgeTriggered) {
    config = config();
    if (!config.isAutoRead()) {
        clearEpollIn0();
        return;
    }
} else {
    config = config();
}
```